src/workerd/api/pyodide/pyodide.h

```diff
@@ -2,15 +2,40 @@
 
 #include <kj/common.h>
 #include <pyodide/pyodide.capnp.h>
+#include <pyodide/generated/pyodide_packages.capnp.h>
 #include <workerd/util/autogate.h>
 
 namespace workerd::api::pyodide {
 
+// A function to read a segment of the tar file into a buffer
+// Set up this way to avoid copying files that aren't accessed.
+class PackagesTarReader: public jsg::Object {
+public:
+  PackagesTarReader() = default;
+
+  void read(jsg::Lock& js, int offset, kj::Array<kj::byte> buf) {
+    auto x = *PYODIDE_PACKAGES_TAR;
+    kj::byte* start = buf.begin();
+    for (int i = 0; i < buf.size(); i++) {
+      start[i] = x[offset + i];
+    }
```

Comment:
Why not use `memcpy`? And for safety should this check that `offset` is within range?

Reply:
How's it look now?